test/unit/mem/view/src/Accessor.cpp

```diff
@@ -0,0 +1,290 @@
+    struct AccessorWithProjection<Projection, MemoryHandle, Elem, BufferIdx, 3>
+    {
+        ALPAKA_FN_ACC auto operator[](alpaka::Vec<alpaka::DimInt<3>, BufferIdx> i) const -> Elem
+        {
+            return Projection{}(accessor[i]);
+        }
+
+        ALPAKA_FN_ACC auto operator()(BufferIdx z, BufferIdx y, BufferIdx x) const -> Elem
+        {
+            return Projection{}(accessor(z, y, x));
+        }
+
+        alpaka::Accessor<MemoryHandle, Elem, BufferIdx, 3, alpaka::ReadAccess> accessor;
+    };
+
+    struct DoubleValue
+    {
+        auto operator()(int i) const
+        {
+            return i * 2;
+        }
+    };
+
+    struct CopyKernel
+    {
+        template<typename Acc, typename Pointer, typename Idx>
+        ALPAKA_FN_ACC void operator()(
+            Acc const&,
+            alpaka::Accessor<Pointer, int, Idx, 1, alpaka::ReadAccess> const src,
+            alpaka::Accessor<Pointer, int, Idx, 1, alpaka::WriteAccess> const dst) const
+        {
+            auto const projSrc = AccessorWithProjection<DoubleValue, Pointer, int, Idx, 1>{src};
+            dst[0] = projSrc[0];
+        }
+    };
+} // namespace
+
+TEST_CASE("projection", "[accessor]")
+{
+    using Dim = alpaka::DimInt<1>;
+    using Size = std::size_t;
+    using Acc = alpaka::ExampleDefaultAcc<Dim, Size>;
+    using DevAcc = alpaka::Dev<Acc>;
+    using Queue = alpaka::Queue<DevAcc, alpaka::Blocking>;
+
+    auto const devAcc = alpaka::getDevByIdx<Acc>(0u);
+    auto queue = Queue{devAcc};
+
+    auto srcBuffer = alpaka::allocBuf<int, Size>(devAcc, Size{1});
+    auto dstBuffer = alpaka::allocBuf<int, Size>(devAcc, Size{1});
+
+    std::array<int, 1> host{{42}};
+    alpaka::memcpy(queue, srcBuffer, host, 1);
+
+    const auto workdiv = alpaka::WorkDivMembers<Dim, Size>{
+        alpaka::Vec<Dim, Size>{Size{1}},
+        alpaka::Vec<Dim, Size>{Size{1}},
+        alpaka::Vec<Dim, Size>{Size{1}}};
+    alpaka::exec<Acc>(queue, workdiv, CopyKernel{}, alpaka::readAccess(srcBuffer), alpaka::writeAccess(dstBuffer));
+
+    alpaka::memcpy(queue, host, dstBuffer, 1);
+
+    REQUIRE(host[0] == 84);
+}
+
+TEST_CASE("constraining", "[accessor]")
+{
+    using Dim = alpaka::DimInt<1>;
+    using Size = std::size_t;
+    using Acc = alpaka::ExampleDefaultAcc<Dim, Size>;
+
+    auto const devAcc = alpaka::getDevByIdx<Acc>(0u);
+    auto buffer = alpaka::allocBuf<int, Size>(devAcc, Size{1});
+
+    alpaka::Accessor<int*, int, Size, 1, std::tuple<alpaka::ReadAccess, alpaka::WriteAccess, alpaka::ReadWriteAccess>>
+        acc = alpaka::accessWith<alpaka::ReadAccess, alpaka::WriteAccess, alpaka::ReadWriteAccess>(buffer);
+
+    alpaka::Accessor<int*, int, Size, 1, alpaka::ReadAccess> readAcc = alpaka::readAccess(acc);
+    alpaka::Accessor<int*, int, Size, 1, alpaka::WriteAccess> writeAcc = alpaka::writeAccess(acc);
+    alpaka::Accessor<int*, int, Size, 1, alpaka::ReadWriteAccess> readWriteAcc = alpaka::access(acc);
```

Comment:
Here is an example how an accessor is further constrained. Currently, I support only constraining a multi-tag accessor into a single tag accessor.